### PR TITLE
Add docker & native build of devfile docs

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
Adds possibility to build devfile docs from json schema using docker or native using special build profiles.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/12524

#### Release Notes
N/A

#### Docs PR
N/A

### CODE REVIEW

File: wsmaster/che-core-api-devfile/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+
+# This is a Dockerfile allowing to build dashboard by using a docker container.
+# Build step: $ docker build -t eclipse-che-dashboard
```
Comment (sleshchenko) [inline wsmaster/che-core-api-devfile/Dockerfile]: Looks like copy/paste error `eclipse-che-dashboard`

File: wsmaster/che-core-api-devfile/pom.xml [hunk 2]
```diff
@@ -191,4 +191,133 @@
             </plugin>
         </plugins>
     </build>
+    <profiles>
+    <profile>
+        <id>devfile-docs-docker</id>
```
Comment (sleshchenko) [inline wsmaster/che-core-api-devfile/pom.xml]: Maybe it would be better to move Devfile docs generation to a separate maven module. WDYT?

Comment (mshaposhnik) [reply wsmaster/che-core-api-devfile/pom.xml]: It anyway will have to contain that profiles (to allow different build types). And they are incactive by default. So we will have module which will do nothing 99% of time.

**Comment by skabashnyuk** [comment]
@mshaposhnik please discuss with @riuvshin do we need maven profilers or not. Maybe the dedicated script is enough. And let us know

**Pull request closed without merging**